src/handler.rs

```diff
@@ -166,6 +167,19 @@ pub struct Response<I, E> {
     item: ResponseTypeItem<I, E>,
 }
 
+impl<I, E> fmt::Debug for Response<I, E> {
+    fn fmt(&self, fmt: &mut fmt::Formatter) -> fmt::Result {
+        let item = format!(
+            "{}",
+            match self.item {
+                ResponseTypeItem::Result(_) => "Result(_)",
+                ResponseTypeItem::Fut(_) => "Fut(_)",
+            }
+        );
+        fmt.debug_struct("Response").field("item", &item).finish()
```

Comment:
I think you can put formatting into `match self.item` instead and avoid allocating temporary `item`

Reply:
It looks creepy. But if you want to, I can change.

Reply:
Not sure I understand, how about to make it like:

```rust

let mut fmt = fmt.debug_struct("Response");

match self.item {
    ResponseTypeItem::Result(_) => fmt.field("item", "Result(_)").finish(),
    ResponseTypeItem::Fut(_) => fmt.field("item", "Fut(_)").finish(),
}
```

I guess there is also problem that debug struct is not the best for printing items like that 😄

Reply:
Well, there is an allocation of fmt :) Is there a difference?
About the way of printing: why not?)

Reply:
>Well, there is an allocation of fmt :) Is there a difference?

Actually there is because you do double allocation ;)

But I guess let's leave it, no one is going to use `Debug` often